### PR TITLE
feat: Enable https in development

### LINKED ISSUE #3: Insecure content warning in development
## Expected Outcome

The extension should load in Contentful when running it with `npm run start`.

## Actual Outcome

The extension does not load and an error message is displayed in the console:

> Blocked loading mixed active content “http://localhost:1234/”

## Steps To Reproduce

1. Create a new extension with the CLI
2. Run `npm run start` 
3. Try to load an entry that contains the UI extension and check the console for errors.

## System information

**Operating system + version:** macOS 10.14.3 beta
**Browser + version:** Firefox Developer Edition 66.0b5

## Screenshots

_Console error:_

screenshot 2019-03-01 23 57 09

_UI Extension:_

screenshot 2019-03-01 23 57 22

### PR DESCRIPTION
**Pull request by connor-baer**
Resolves #3.

## Changes

- Enable `https` on the parcel dev server

### CODE REVIEW

**Comment by suevalov** [comment]
@connor-baer I was thinking to enable `https` as well, but make it optional. 

Would it make sense for you to use something like:
```
const https = process.env.HTTPS === 'true';
```

and just write additional documentation how to use this instead of modifying `npm run start` script?

File: packages/contentful-extension-scripts/scripts/start.js [hunk 1]
```diff
@@ -19,17 +19,19 @@ const options = {
   scopeHoist: false, // Turn on experimental scope hoisting/tree shaking flag, for smaller production bundles
   logLevel: 3, // 5 = save everything to a file, 4 = like 3, but with timestamps and additionally log http requests to dev server, 3 = log info, warnings & errors, 2 = log warnings & errors, 1 = log errors
   hmr: true, // Enable or disable HMR while watching
-  hmrPort: 0, // The port the HMR socket runs on, defaults to a random free port (0 in node.js resolves to a random free port)
+  hmrPort: 54321, // The port the HMR socket runs on, defaults to a random free port (0 in node.js resolves to a random free port)
```
Comment (suevalov) [inline packages/contentful-extension-scripts/scripts/start.js]: I believe that part of the comment is redundant after the change: ```defaults to a random free port (0 in node.js resolves to a random free port)```

Comment (connor-baer) [reply packages/contentful-extension-scripts/scripts/start.js]: True! I'll update it.

**Comment by connor-baer** [comment]
@suevalov I'm open to making `https` configurable, especially since it's a bit tricky to configure in the browsers. 

However, I assume that most people will need it since the latest versions of Firefox and Chrome block any insecure iframe content if the parent page is served over `https`.

**Comment by suevalov** [comment]
@connor-baer I'd go with `http` as a default because we're about to release a small UX improvement in the web app, so when an extension is served from `http://localhost` there will be a small note about a need of enabling `insecure iframe content` in the browser for proper development mode.

`https` is a very nice addition, but as you said it can be tricky to configure.

**Comment by connor-baer** [comment]
@suevalov Alright! I've made the requested changes, can you review again please? :)

**Comment by suevalov** [comment]
@connor-baer Available in `@contentful/contentful-extension-scripts@0.3.0`

**Comment by connor-baer** [comment]
> [...] there will be a small note about a need of enabling `insecure iframe content` in the browser for proper development mode.

I've been thinking about this — by advising developers to allow insecure iframe content, you're undermining their security and all the progress that's been made in moving the web towards full encryption ☹️ 

I believe it's better to add an exception for an individual self-signed certificate, one, because it's more secure, and two, because it's little more effort than changing your browser settings. 

Would you consider to switch on https by default?

**Comment by suevalov** [comment]
@connor-baer In Chrome `load insecure script` is enable only for the current session and automatically disabled when you close browser, so it's not a big issue in terms of security I believe. 

Nevertheless, I'll discuss it with the team and we might change it.

**Pull request merged**